From 9783519ae3cf90ed5871b04dd84ec39985b5eb17 Mon Sep 17 00:00:00 2001
From: Richard Fitzgerald <rf@opensource.wolfsonmicro.com>
Date: Thu, 5 Jun 2014 15:16:42 +0100
Subject: [PATCH 220/351] mfd: arizona: zero-init the out-mono DT array

The earlier change to read out_mono pdata from DT wasn't
zero-initializing the array so in the event that the DT
didn't contain an out-mono setting, the pdata would be
filled in from garbage.

Change-Id: Ie982d87a0d8eea9cb2cd8513b4331715c06279f0
Signed-off-by: Richard Fitzgerald <rf@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/mfd/arizona-core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mfd/arizona-core.c b/drivers/mfd/arizona-core.c
index effb871..e049687 100644
--- a/drivers/mfd/arizona-core.c
+++ b/drivers/mfd/arizona-core.c
@@ -841,7 +841,7 @@ static int arizona_of_get_micbias(struct arizona *arizona,
 static int arizona_of_get_core_pdata(struct arizona *arizona)
 {
 	struct arizona_pdata *pdata = &arizona->pdata;
-	u32 out_mono[ARRAY_SIZE(pdata->out_mono)];
+	u32 out_mono[ARRAY_SIZE(pdata->out_mono)] = {0};
 	int i;
 
 	pdata->reset = arizona_of_get_named_gpio(arizona, "wlf,reset", true);
-- 
1.7.9.5

